cmd/cli/serve/util.go

```diff
@@ -167,16 +169,29 @@ func getAllowListedLocalPathsConfig() []string {
 	return viper.GetStringSlice(types.NodeAllowListedLocalPaths)
 }
 
-func getNetworkConfig() (node.NetworkConfig, error) {
+func getNetworkConfig(nodeID string) (node.NetworkConfig, error) {
 	var networkCfg types.NetworkConfig
 	if err := config.ForKey(types.NodeNetwork, &networkCfg); err != nil {
 		return node.NetworkConfig{}, err
 	}
+
+	// Generate an auth token using the user's client key. It should not be
+	// possible to compute this value by anyone other than the NATS server, and
+	// should be stable across restarts of the node.
+	var keySig string
+	keySig, err := system.SignForClient([]byte(nodeID))
+	if err != nil {
+		return node.NetworkConfig{}, err
+	}
+	hash := sha256.Sum256([]byte(keySig))
+	token := base64.RawURLEncoding.EncodeToString(hash[:])
```

Comment:
@frrist – note that this token is auto-generated by the node on boot, but is predictable. Does this cause a problem for marketplace work?